### PR TITLE
do not claim any disk that has non-Swift role assigned

### PR DESCRIPTION
**Pull request by jsuchome**
This is based on disk_roles setting added by https://github.com/crowbar/barclamp-crowbar/pull/1306 and  crowbar/barclamp-deployer#248

### CODE REVIEW

**Comment by vuntz** [comment]
+1, but I understand code might change after discussion in https://github.com/crowbar/barclamp-deployer/pull/248

File: chef/cookbooks/swift/recipes/disks.rb [hunk 1]
```diff
@@ -30,12 +30,14 @@ def get_uuid(disk)
...
-unclaimed_disks.each do |k|
-  to_use_disks << k
-end
-claimed_disks = BarclampLibrary::Barclamp::Inventory::Disk.claimed(node, "Swift")
+claim_string = "Swift"
+
+# take only disks with Swift role; when there is none, take disks without role
+swift_disks, disks_without_role =
+  BarclampLibrary::Barclamp::Inventory::Disk.prepared_for(node, claim_string)
+to_use_disks = swift_disks.empty? ? disks_without_role : swift_disks
```
Comment (jsuchome) [inline chef/cookbooks/swift/recipes/disks.rb]: Hm, does this mean, that in the next chef run (after disk with swift role was already used), disk with no role will be taken anyway? That doesn't seem to help...

**Comment by rsalevsky** [comment]
@jsuchome, this repository got merged into [crowbar-openstack](https://github.com/crowbar/crowbar-openstack).
Please open a new pull request there. __Do NOT merge here anymore!__

**Comment by jsuchome** [comment]
Closing, as there's already PR to new repo

**Pull request closed without merging**